base/ltcmd.dtx

```diff
@@ -435,18 +435,75 @@
 %    \end{macrocode}
 % \end{macro}
 %
+% \begin{macro}[EXP]{\@@_all_m_check:n, \@@_all_m_check_aux:n}
+%   A quick loop to check for all |(+)m|-type arguments.
+%    \begin{macrocode}
+\cs_new:Npn \@@_all_m_check:n #1
+  { \tl_map_function:nN {#1} \@@_all_m_check_aux:n }
+\cs_new:Npn \@@_all_m_check_aux:n #1
+  {
+    \str_if_eq:nnF {#1} { m }
+      {
+        \str_if_eq:nnF {#1} { + }
+          { X }
+      }
+  }
+%    \end{macrocode}
+% \end{macro}
+%
 % \begin{macro}{\@@_declare_cmd_code:Nnn}
 % \begin{macro}
-%   {\@@_declare_cmd_code_aux:Nnn, \@@_declare_cmd_code_expandable:Nnn}
-%   The appropriate auxiliary is called.
+%   {
+%     \@@_declare_cmd_optimised:Nnn,
+%     \@@_declare_cmd_code_aux:Nnn,
+%     \@@_declare_cmd_code_expandable:Nnn
+%   }
+% \begin{macro}{\@@_start_optimised:}
+% \changes{v1.2b}{2023/11/28}
+%   {Optimise cmd creation for all-\texttt{m} arguments}
+%   At this stage we can check for a short-cut possibility: if the signature
+%   is just made up of \cs{@@_expandable_grab_m:w}, then we can produce an
```

Comment:
or `\cs{@@_expandable_grab_long_m:w}`

(Then ambiguity arises because the combination of short and long `m` are not supported. Fine, code comment won't be executed.)

Reply:
I've updated the comment